src/config/config.ts

```diff
@@ -1,210 +1,106 @@
+// Construct bucket URLs depending on node environment
+// If in development env, endpoint communicates with localstack, a fully
+// functional local AWS cloud stack for hosting images/logos/attachments.
+// Else, the environment variables to instantiate S3 are used.
 
-  logger.error({
-    message: errMsg,
-    meta: {
-      action: 'init',
-    },
+// Perform validation before accessing s3 Bucket Urls
+const s3BucketUrlSchema = loadS3BucketUrlSchema({
+  isDev,
+  region: basicVars.awsConfig.region,
+})
+const awsEndpoint = convict(s3BucketUrlSchema).getProperties().endPoint
+const s3BucketUrlVars = convict(s3BucketUrlSchema)
+  .load({
+    logoBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.logoS3Bucket}`,
+    imageBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.imageS3Bucket}`,
+    // NOTE THE TRAILING / AT THE END OF THIS URL! This is only for attachments!
+    attachmentBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.attachmentS3Bucket}/`,
   })
-  throw new Error(errMsg)
+  .validate({ allowed: 'strict' })
+  .getProperties()
+
+const s3 = new aws.S3({
+  region: basicVars.awsConfig.region,
+  // Unset and use default if not in development mode
+  // Endpoint and path style overrides are needed only in development mode for
+  // localstack to work.
+  endpoint: isDev ? s3BucketUrlVars.endPoint : undefined,
+  s3ForcePathStyle: isDev ? true : undefined,
+})
+
+const awsConfig: AwsConfig = {
+  ...s3BucketUrlVars,
+  ...basicVars.awsConfig,
+  s3,
 }
 
-// Optional environment variables
-/**
- * Name of CloudWatch log group to store short-term logs. Log streams are
- * separated by date.
- */
-const customCloudWatchGroup = process.env.CUSTOM_CLOUDWATCH_LOG_GROUP
-
-/**
- * Load env variable with General Maintenance banner text.
- */
-const isGeneralMaintenance = process.env.IS_GENERAL_MAINTENANCE
-
-/**
- * The banner message on login page. Allows for HTML.
- */
-const isLoginBanner = process.env.IS_LOGIN_BANNER
-
-/**
- * The banner message to show on all pages. Allows for HTML. Will supersede
- * all other banner content if it exists.
- */
-const siteBannerContent = process.env.SITE_BANNER_CONTENT
-
-/**
- * The banner message to show on on admin pages. Allows for HTML.
- */
-const adminBannerContent = process.env.ADMIN_BANNER_CONTENT
-
-// Configs
-const appConfig: AppConfig = {
-  title: process.env.APP_NAME || defaults.app.name,
-  description: process.env.APP_DESC || defaults.app.desc,
-  appUrl: process.env.APP_URL || defaults.app.url,
-  keywords: process.env.APP_KEYWORDS || defaults.app.keywords,
-  images: (process.env.APP_IMAGES || defaults.app.images).split(','),
-  twitterImage: process.env.APP_TWITTER_IMAGE || defaults.app.twitterImage,
+let dbUri
+if (isDev) {
+  if (basicVars.core.nodeEnv === Environment.Dev && prodOnlyVars.dbHost) {
+    dbUri = prodOnlyVars.dbHost
+  } else if (basicVars.core.nodeEnv === Environment.Test) {
+    dbUri = undefined
+  } else {
+    throw new Error('Database configuration missing')
```

Comment:
not sure if this case is necessary, if `isDev` is truthy then one of the earlier two cases must be true

Reply:
it is possible. 
`if (basicVars.core.nodeEnv === Environment.Dev && prodOnlyVars.dbHost)` => will be false if `prodOnlyVars.dbHost` is `undefined`

Reply:
ah yes! my bad